app/controllers/products_controller.rb

```diff
@@ -0,0 +1,78 @@
+class ProductsController < ApplicationController
+  before_action :find_product, only: [:show, :edit, :update, :destroy]
+
+  def index
+      @products= Product.all
+      searchFilter
+  end
+
+  def show
+  end
+
+  def new
+      @product = Product.new
+  end
+
+  def create
+    @product = Product.new(product_params)
+    if @product.save
+        flash[:success] = "Successfully Added"
+        redirect_to products_path
+    else
+        render :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @product.update(product_params)
+      flash[:success] = "Successfully Updated"
+      redirect_to products_path
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    if @product.destroy
+      redirect_to products_path, notice: "Product was successfully destroyed."
+    else
+      redirect_to products_path, alert: "Unable to destroy Product."
+    end
+  end
+
+  private
+
+  def product_params
+      params.require(:product).permit(:title, :description, :price, :total_quantity, :quantity_in_stock,:display_picture)
+  end
+
+  def searchFilter
+    if params[:search].present?
+      @products = @products.where("title ILIKE ?", "%#{params[:search]}%")
+    end
+
+    if params[:min_price].present? && params[:max_price].present?
```

Comment:
I Guess params[:min_price].present? || params[:max_price].present? we can use this condition here and we don't need these two methods

if params[:min_price].present?
      min_price = params[:min_price]
      @products = @products.where("price >= ?", min_price)
    end

    if params[:max_price].present?
      max_price = params[:max_price]
      @products = @products.where("price <= ?", max_price)
    end